pandas/core/arrays/categorical.py

```diff
@@ -1343,16 +1353,15 @@ def searchsorted(self, value, side='left', sorter=None):
                              ".as_ordered() to change the Categorical to an "
                              "ordered one")
 
-        from pandas.core.series import Series
-
-        values_as_codes = _get_codes_for_values(Series(value).values,
-                                                self.categories)
-
-        if -1 in values_as_codes:
-            raise ValueError("Value(s) to be inserted must be in categories.")
+        if is_scalar(value):
```

Comment:
i think this is confusing code because get_loc raises, i would rather just use .get_indexer here

Reply:
The point of searchsorted is fast searching. ``get_indexer`` is currently very very slow, as it always creates an array. ``get_loc`` OTOH can return scalar or a slice, which is both faster to create and faster to use.

So I think we need to keep ``get_loc``, unless ``get_indexer`` gets a redesign

Reply:
and this actually makes a difference? show this specific case

i am sure that optimizing get_indexer would not be hard and is a better soln

Reply:
```python
>>> c = pd.Categorical(list('a' + 'b'  + 'c' ))
>>> %timeit c.categories.get_loc('b')
1.19 µs
>>> %timeit c.categories.get_indexer(['b'])
261 µs
```

I can take  look at optimizing get_indexer